src/core/currency/wallet/currency-wallet-reducer.ts

```diff
@@ -425,17 +428,21 @@ export function mergeTx(
   defaultCurrency: string,
   oldTx: MergedTransaction = defaultTx
 ): MergedTransaction {
-  const currencyCode =
-    tx.currencyCode != null ? tx.currencyCode : defaultCurrency
+  const {
+    currencyCode = defaultCurrency,
+    isSend = gt(tx.nativeAmount, '0') || gt(tx.parentNetworkFee ?? '0', '0')
```

Comment:
Agreed. As well, resist the temptation to look at tx.networkFee since that might also be set for a receive transaction. Only look at whether `nativeAmount` is < 0. Only the plugin can determine if the user actually spent the funds for this tx.